File: task-3/Haskell/Sub-Task 4.hs
main :: IO ()
main = do
    n <- readFile "input.txt"
    writeFile "output.txt" (unlines (diamond (read n :: Int)))

diamond :: Int -> [String]
diamond n = top ++ bottom
  where
    top = [replicate (n - i - 1) ' ' ++ replicate (2 * i + 1) '*' | i <- [0 .. n - 1]]
    bottom = reverse (init top)

File: task-3/C++/Sub-Task 4.cpp
#include <fstream>
#include <string>

int main() {
    std::ifstream infile("input.txt");
    std::ofstream outfile("output.txt");
    int n;
    infile >> n;
    for (int i = 0; i < n; i++) {
        outfile << std::string(n - i - 1, ' ') << std::string(2 * i + 1, '*') << std::endl;
    }
    for (int i = n - 2; i >= 0; i--) {
        outfile << std::string(n - i - 1, ' ') << std::string(2 * i + 1, '*') << std::endl;
    }
    infile.close();
    outfile.close();
    return 0;
}
